libs/common/src/services/event/event-collection.service.ts

```diff
@@ -9,17 +9,71 @@ import { EventType } from "../../enums";
 import { EventData } from "../../models/data/event.data";
 import { StateProvider } from "../../platform/state";
 import { CipherService } from "../../vault/abstractions/cipher.service";
+import { CipherView } from "../../vault/models/view/cipher.view";
 
 import { EVENT_COLLECTION } from "./key-definitions";
 
 export class EventCollectionService implements EventCollectionServiceAbstraction {
+  private orgIds$: Observable<string[]>;
+
   constructor(
     private cipherService: CipherService,
     private stateProvider: StateProvider,
     private organizationService: OrganizationService,
     private eventUploadService: EventUploadService,
     private authService: AuthService,
-  ) {}
+  ) {
+    this.orgIds$ = this.organizationService.organizations$.pipe(
+      map((orgs) => orgs?.filter((o) => o.useEvents)?.map((x) => x.id) ?? []),
+    );
+  }
+
+  /** Adds an event to the active user's event collection
+   *  @param eventType the event type to be added
+   *  @param ciphers The collection of ciphers to log events for
+   *  @param uploadImmediately in some cases the recorded events should be uploaded right after being added
+   */
+  async collectMany(
+    eventType: EventType,
+    ciphers: CipherView[],
+    uploadImmediately = false,
+  ): Promise<any> {
+    const userId = await firstValueFrom(this.stateProvider.activeUserId$);
+    const eventStore = this.stateProvider.getUser(userId, EVENT_COLLECTION);
+
+    if (!(await this.shouldUpdate(null, eventType, ciphers))) {
+      return;
+    }
+
+    const events$ = this.orgIds$.pipe(
+      map((orgs) => {
+        const filtered = ciphers.filter((c) => orgs.includes(c.organizationId));
+        return filtered.map((c) => {
+          const event = new EventData();
+          event.type = eventType;
+          event.cipherId = c.id;
+          event.date = new Date().toISOString();
+          event.organizationId = c.organizationId;
+          return event;
+        });
+      }),
+    );
```

Comment:
⛏️ I believe this could be streamlined into something like
```suggestion
    const events$ = this.orgIds$.pipe(
      map((orgs) => 
        ciphers
          .filter((c) => orgs.includes(c.organizationId))
          .map((c) => ({
            type: eventType,
            cipherId: c.id,
            date: new Date().toISOString(),
            organizationId: c.organizationId
          }))
      )
    );
```